starfish/pipeline/features/spots/detector/_base.py

```diff
@@ -1,9 +1,30 @@
+from typing import Union
+
+import xarray as xr
+import numpy as np
+
 from starfish.image import ImageStack
 from starfish.pipeline.algorithmbase import AlgorithmBase
+from starfish.pipeline.features.spot_attributes import SpotAttributes
+from starfish.intensity_table import IntensityTable
 
 
 class SpotFinderAlgorithmBase(AlgorithmBase):
-    # TODO ambrosejcarr: should return IntensityTable when refactor is complete
-    def find(self, hybridization_image: ImageStack):
-        """Find spots."""
+    def find(self, hybridization_image: ImageStack, blobs_image: Union[np.ndarray, xr.DataArray]) \
+            -> IntensityTable:
+        """Finds spots in an ImageStack"""
         raise NotImplementedError()
+
+    def image_to_spots(self, data_image: Union[np.ndarray, xr.DataArray]) -> SpotAttributes:
+        """Finds spots in a 3d volume"""
+        raise NotImplementedError()
+
+    @staticmethod
+    def _get_measurement_function(measurement_type: str):
```

Comment:
why does this not return Callable[[Sequence], Number] like you're doing in `measure_spot_intensities`?